### PR TITLE
Feature/check intersections of shapes

### PR DESCRIPTION
**Pull request by PetervDooren**
Merge after geolib2 https://github.com/tue-robotics/geolib2/pull/34

### CODE REVIEW

File: src/ed.cpp [hunk 1]
```diff
@@ -388,17 +388,35 @@ bool srvSimpleQuery(ed_msgs::SimpleQuery::Request& req, ed_msgs::SimpleQuery::Re
...
+                else
+                {
+                    geom_ok = (radius * radius > (e->pose().t - center_point).length2());
+                }
+            }
+            else if (shape)
+            {
+                geo::Vector3 center_point_e = e->pose().getBasis().transpose() * (center_point - e->pose().getOrigin());
+                geom_ok = shape->contains(center_point_e);
+            }
```
Comment (MatthijsBurgh) [inline src/ed.cpp]: Would the code be less complex if first you do `if (shape)` and then do `if (radius > 0)`?

File: src/ed.cpp [hunk 2]
```diff
@@ -388,17 +388,35 @@ bool srvSimpleQuery(ed_msgs::SimpleQuery::Request& req, ed_msgs::SimpleQuery::Re
...
-        if (geom_ok)
-        {
-            res.entities.push_back(ed_msgs::EntityInfo());
-            convert(*e, res.entities.back());
+            if (!geom_ok)
+                continue;
         }
+
+        res.entities.push_back(ed_msgs::EntityInfo());
+        convert(*e, res.entities.back());
```
Comment (MatthijsBurgh) [inline src/ed.cpp]: Is it correct this is outside the `if (radius < std::numeric_limits<double>::infinity())`??

File: src/ed.cpp (hunk 1 above)
Comment (PetervDooren) [reply src/ed.cpp]: No

File: src/ed.cpp (hunk 2 above)
Comment (PetervDooren) [reply src/ed.cpp]: Yes

File: src/ed.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply src/ed.cpp]: I will take another look at it today.

Comment (MatthijsBurgh) [reply src/ed.cpp]: See 89f4b08509d7108fbb1a28a49fb49c1a2f12718f

**Pull request merged**